src/Tests/Generation/GirLoader.Tests/DisposableTempDirectory.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.IO;
+
+namespace GirLoader.Test;
+
+public class DisposableTempDirectory : IDisposable
+{
+    private readonly string _directoryPath;
+
+    public DisposableTempDirectory()
+    {
+        _directoryPath = Path.Combine(Path.GetTempPath(), Path.GetRandomFileName());
+        Directory.CreateDirectory(_directoryPath);
+    }
+
+    public string DirectoryPath => _directoryPath;
```

Comment:
I had called this `DirectoryPath` rather than `Path` to avoid clashing with the `System.IO.Path` namespace, but I've renamed this and updated the other uses of `Path` to qualify them with their namespace.